src/axom/mir/docs/sphinx/mir_algorithms.rst

```diff
@@ -0,0 +1,153 @@
+
+Axom's implementation of Equi-Z is data parallel and can run on the CPU and the GPU.
+First, the zones of interest are identified and they are classified as clean or mixed.
+Clean consist of a single material and are pulled out early into a new mesh while mixed
+zones are sent into the Equi-Z algorithm to reconstruct zones where material interfaces
+exist. The two meshes are finally merged to form a single output mesh. The mesh may
+consist of multiple Blueprint shape types in an unstructured "mixed" topology.
+
+Axom's implementation supports 2D/3D zones from structured or unstructured topologies
+made of Finite Element Zoo elements *(e.g. triangles, quadrilaterals, tetrahedra, pyramids,
+wedges, hexahedra, or topologically-compatible mixtures)*. The MIR logic for Equi-Z is
+encapsulated in ``axom::mir::EquizAlgorithm``, which is a class that is templated on view objects.
+View objects help provide an interface between the Blueprint data and the MIR algorithm.
+At a minimum, an execution space and three views are required to instantiate the
+``axom::mir::EquiZAlgorithm`` class. The execution space determines which compute backend will
+be used to execute the algorithm. The Blueprint data must exist in a compatible
+memory space for the execution space. The views are: *CoordsetView*, *TopologyView*, and
+*MaterialView*. The *CoordsetView* template argument lets the algorithm access the mesh's
+coordset using concrete data types and supports queries that return points. The
+*TopologyView* provides a set of operations that can be performed on meshes, mainly a
+method for retrieving individual zones that can be used in device kernels.
+The *MaterialView* provides an interface for querying matsets.
+
+Once view types have been created and views have been instantiated, the ``EquiZAlgorithm``
+algorithm can be instantiated and used. The ``EquiZAlgorithm`` class provides a single
+``execute()`` method that takes the input mesh, an options node, and a node to contain
+the output mesh. The output mesh will exist in the same memory space as the input mesh,
+which again, must be compatible with the selected execution space. The ``axom::mir::utilities::blueprint::copy()``
+function can be used to copy Conduit nodes from one memory space to another.
+
+.. literalinclude:: ../../examples/concentric_circles/runMIR.hpp
+   :start-after: _equiz_mir_start
+   :end-before: _equiz_mir_end
+   :language: C++
+
+The MIR output will contain a new field called *"originalElements"* that indicates which
+original zone number gave rise to the reconstructed zone. This field makes it possible
+to map back to the original mesh. The name of the field can be changed using options.
+
+#####################
+Example Application
+#####################
+
+The mir_concentric_circles application generates a uniform mesh populated with circular
+mixed material shells and then it performs MIR on the input mesh before writing the
+reconstructed mesh.
+
```

Comment:
Which arguments are required? What are the defaults?

Reply:
I added some defaults, and a sentence indicating all arguments are optional.